### PR TITLE
Add ability to use SQL to filter pings included in generated heatmaps

### PR DESCRIPTION
**Pull request by JoelClemence**
* Create FilterQuerySqlRepository to facilitate retrieval of SQL files
* Add parameter to job

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ This project provides a simple Terraform plugin to Gradle.
 
 ## Running heatmap generation via step function console
 
-To orchestrate the generation of heatmaps with both combinations of resampling parameters, a step function is used which passes the `year`, `month`, `prefix` and `output` parameters through to the Batch job.
+To orchestrate the generation of heatmaps with both combinations of resampling parameters, a step function is used which passes the `filter_sql`,`year`, `month`, `prefix` and `output` parameters through to the Batch job.
```
Comment (galexite) [inline README.md]: Space after `filter_sql` comma?

Comment (JoelClemence) [reply README.md]: Done

File: deployment/src/main/terraform/modules/batch/orchestration.tf [hunk 2]
```diff
@@ -74,7 +75,8 @@ resource aws_sfn_state_machine batch_heatmap_step_fn {
                   "year.$": "$.year",
                   "month.$": "$.month",
                   "prefix.$": "$.prefix",
-                  "output.$": "$.output"
+                  "output.$": "$.output",
+                  "filter_sql.$": "$.filter_sql"
```
Comment (andyphelps) [inline deployment/src/main/terraform/modules/batch/orchestration.tf]: Can we get consistency of naming with filter-sql and filter-sql-filename.  "filter-sql" implies SQL not a file containing sql.

File: heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/ConfigParser.scala [hunk 3]
```diff
@@ -48,6 +48,11 @@ object ConfigParser {
         .valueName("<output directory/S3 bucket>")
         .text("location to output the resulting GeoTIFF and PNG")
         .action((value, config) => config.copy(outputDirectory = value)),
+      opt[String]('f', "filter-sql")
```
Comment (andyphelps) [inline heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/ConfigParser.scala]: as previous, naming consistency

File: heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/repository/FilterQuerySqlRepository.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    IOUtils
+      .readLines(fileInputStream)
+      .toArray
+      .mkString("\n")
+  }
+
+  private def getS3FileInputStream(sqlFile: String): InputStream = {
+    val s3Uri: AmazonS3URI = new AmazonS3URI(sqlFile)
+    s3Client.getObject(s3Uri.getBucket, s3Uri.getKey).getObjectContent
+  }
```
Comment (andyphelps) [inline heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/repository/FilterQuerySqlRepository.scala]: Note sure I like seeing code specifically to allow local testing in here.  You could mock the AWS sdk?

File: heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/service/GeoTiffS3KeyService.scala [hunk 5]
```diff
@@ -4,23 +4,27 @@ import java.time.LocalDate
...
+    val objectPrefix = s"sqlFilename=$sqlFilenameDescriptor.sql/resample=$interpolationDescriptor/type=monthly" +
       s"/year=${config.year}/month=${config.month}/"
     val filename =
       s"$sqlFilenameDescriptor-1km-res-$interpolationDescriptor-monthly-$dateDescriptor.tif"
 
     s"$objectPrefix$filename"
   }
 
+  private def createSqlFilenameDescriptor(config: Config): String =
+    FilenameUtils.getBaseName(config.filterSqlFile.split("/").last)
```
Comment (andyphelps) [inline heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/service/GeoTiffS3KeyService.scala]: Do we want to strip the extension also?

Comment (JoelClemence) [reply heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/service/GeoTiffS3KeyService.scala]: Yes - the filename of the SQL file is included in the filename of the generted TIF file and may be strange to see something like ```unfiltered.sql-1km-res-6hr-30km-monthly-Jan-2019.tf``` unless this is preferable, might be worth a knock-knock?

File: heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/repository/FilterQuerySqlRepository.scala (hunk 4 above)
Comment (JoelClemence) [reply heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/repository/FilterQuerySqlRepository.scala]: Added unit test to cover S3 path

File: heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/Main.scala [hunk 6]
```diff
@@ -1,11 +1,12 @@
 package uk.gov.ukho.ais.heatmaps
 
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3Client}
 import uk.gov.ukho.ais.heatmaps.service.AthenaDataSourceProvider
-import javax.sql.DataSource
 
 object Main {
 
   def main(args: Array[String]): Unit = {
+    implicit val amazonS3: AmazonS3 = AmazonS3Client.builder().build()
```
Comment (andyphelps) [inline heatmapGenerator/src/main/scala/uk/gov/ukho/ais/heatmaps/Main.scala]: AmazonS3Client.builder().build() can be
AmazonS3ClientBuilder.defaultClient()

**Pull request merged**